community/kernel/src/main/java/org/neo4j/helpers/Settings.java

```diff
@@ -47,10 +47,13 @@
  * which means that you don't want any default value at all, and MANDATORY, which means that the user has to specify a value
  * for this setting. Not providing a mandatory value for a setting leads to an IllegalArgumentException.
  *
- * If a setting does not have a provided value, and no default, then
+ * @deprecated this class is deprecated and will be moved to internal packages in the next major release
  */
+@Deprecated
```

Comment:
why do we need to deprecate this class and introduce new one and wait for 4.0 instead of just move class to different package?